radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixJsonRpcClient.java

```diff
@@ -272,7 +272,9 @@ public boolean tryClose() {
 						} else {
 							message = null;
 						}
-						return AtomSubmissionUpdate.now(atom.getHid(), state, message);
+						AtomSubmissionUpdate update = AtomSubmissionUpdate.create(atom.getHid(), state, message);
+						update.putMetaData("jsonRpcParams", params);
```

Comment:
Should this be here?  Not sure what the long-term intent of metadata here is for?
Does this only appear in JSON output, or is there another planned use?

Reply:
Structure and design probably needs to be refactored but the idea is to be able to pass up lower level information (e.g. time-received, serialized-object) to higher level.